README.md
# CasaDeMarcat

RO  
 Aplicatie pentru casele de marcat V0.80  

BUG-uri cunoscute  
 -La plata cu tichete se poate introduce o valoare mai mare decat cea ce trebuie platita(importanta medie)  

MODIFICARI  
 -0.76 In loc de instantieri ale unei clase, acum programul preia produsele din baza de date  
 -0.76 Butoanele de plata cash si cel de " alta suma" acum functioneaza corespunzator   
 -0.76 Am adaugat posibilitatea de a plati cu bonuri, card, cash, toate  pe acelasi bon  

 -0.80 Generare unui fisier pdf in care se afla bonul pentru printare  
 -0.80 Bonul este salvat in Baza de date    
 -0.80 Implementare functionalitate rest  
 -0.80 Butonul "TOTAL" din lista cu optiuni acum va sterge toate datele din tabel daca in acesta sunt doar produse sterse cu valoarea totala 0    
 -0.80 Tabelul principal este acum extins dinamic( linii inceput: 47 - pentru a oferi un aspect vizual placut)  

--------------------------------------------------------------------------------------------------------------
--------------------------------------------------------------------------------------------------------------
--------------------------------------------------------------------------------------------------------------

EN  
 Cashier app V0.80  
KNOWN BUGS  
 -The current state of the app allows the user to pay more than they have to(ticket payment)  

CHANGES  
 -0.76 Items are now loaded from  SQL instead of class instances   
 -0.76 All cash payment buttons now work as intended    
 -0.76 Now the customer can pay with tickets, cash and card, all 3 at once   

 -0.80 A pdf receipt file is now generated in  "Bonuri" , allowing it to be printed    
 -0.80 The receipt is also saved in the database    
 -0.80 Implemented the "give change" functionality    
 -0.80 The "TOTAL" buton will now erase the table if there are only cancelled items and their total is 0    
 -0.80 Main table is now dinamically extended(At first there are only 47 line just to make sure it's visually pleasing)    



Changelog.md
# CasaDeMarcat

RO
 Aplicatie pentru casele de marcat v.80
 
BUG-uri cunoscute
 La plata cu tichete se poate introduce o valoare mai mare decat cea ce trebuie platita(importanta medie)

MODIFICARI
 0.76 In loc de instantieri ale unei clase, acum programul preia produsele din baza de date.
 0.76 Butoanele de plata cash si cel de " alta suma" acum functioneaza corespunzator.
 0.76 Am adaugat posibilitatea de a plati cu bonuri, card, cash, toate  pe acelasi bon.

 0.80 Generare unui fisier pdf in care se afla bonul pentru printare.
 0.80 Bonul este salvat in Baza de date.
 0.80 Implementare functionalitate rest.
 0.80 Butonul "TOTAL" din lista cu optiuni acum va sterge toate datele din tabel daca in acesta sunt doar produse sterse cu valoarea totala 0.
 0.80 Tabelul principal este acum extins dinamic.( linii inceput: 47 - pentru a oferi un aspect vizual placut)

--------------------------------------------------------------------------------------------------------------
--------------------------------------------------------------------------------------------------------------
--------------------------------------------------------------------------------------------------------------

EN
 Cashier app V 0.80
KNOWN BUGS
 The current state of the app allows the user to pay more than they have to(ticket payment).

CHANGES
 0.76 Items are now loaded from  SQL instead of class instances.
 0.76 All cash payment buttons now work as intended.
 0.76 Now the customer can pay with tickets, cash and card, all 3 at once.

 0.80 A pdf receipt file is now generated in  "Bonuri" , allowing it to be printed.
 0.80 The receipt is also saved in the database.
 0.80 Implemented the "give change" functionality.
 0.80 The "TOTAL" buton will now erase the table if there are only cancelled items and their total is 0.
 0.80 Main table is now dinamically extended(At first there are only 47 line just to make sure it's visually pleasing).



src/main/java/com/mycompany/casademarcat/previewBon.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


import com.mycompany.casademarcat.login;
import static com.mycompany.casademarcat.login.loggedPass;
import static com.mycompany.casademarcat.login.loggedUser;
import com.mycompany.casademarcat.loginAproved;
import static com.mycompany.casademarcat.loginAproved.jTable1;
import static com.mycompany.casademarcat.loginAproved.nrBon;
import static com.mycompany.casademarcat.loginAproved.rest;
import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.util.Date;




/**
 *
 * @author kodie
 */
public class previewBon {
    
    int nrBon;
    int nrBuc;
    double rest;
    double pretUnitar;
    double pretTotal;
    String numeProdus;
    String casier;
    String metodaDePlata1; 
    String metodaDePlata2;
    String metodaDePlata3;
    String timp;

    public previewBon(int nrBon, int nrBuc, double rest, double pretUnitar, double pretTotal, String numeProdus, String casier, String metodaDePlata1, String metodaDePlata2, String metodaDePlata3, String timp) {
        this.nrBon = nrBon;
        this.nrBuc = nrBuc;
        this.rest = rest;
        this.pretUnitar = pretUnitar;
        this.pretTotal = pretTotal;
        this.numeProdus = numeProdus;
        this.casier = casier;
        this.metodaDePlata1 = metodaDePlata1;
        this.metodaDePlata2 = metodaDePlata2;
        this.metodaDePlata3 = metodaDePlata3;
        this.timp = timp;
        
 
    }

    public previewBon(int nrBon, int nrBuc, double rest, double pretUnitar, double pretTotal, String numeProdus, String casier, String metodaDePlata1, String metodaDePlata2, String timp) {
        this.nrBon = nrBon;
        this.nrBuc = nrBuc;
        this.rest = rest;
        this.pretUnitar = pretUnitar;
        this.pretTotal = pretTotal;
        this.numeProdus = numeProdus;
        this.casier = casier;
        this.metodaDePlata1 = metodaDePlata1;
        this.metodaDePlata2 = metodaDePlata2;
        this.timp = timp;
    }

    public previewBon(int nrBon, int nrBuc, double rest, double pretUnitar, double pretTotal, String numeProdus, String casier, String metodaDePlata1, String timp) {
        this.nrBon = nrBon;
        this.nrBuc = nrBuc;
        this.rest = rest;
        this.pretUnitar = pretUnitar;
        this.pretTotal = pretTotal;
        this.numeProdus = numeProdus;
        this.casier = casier;
        this.metodaDePlata1 = metodaDePlata1;
        this.timp = timp;
    }

    public void setNrBon(int nrBon) {
        this.nrBon = nrBon;
    }

    public void setNrBuc(int nrBuc) {
        this.nrBuc = nrBuc;
    }

    public void setRest(double rest) {
        this.rest = rest;
    }

    public void setPretUnitar(double pretUnitar) {
        this.pretUnitar = pretUnitar;
    }

    public void setPretTotal(double pretTotal) {
        this.pretTotal = pretTotal;
    }

    public void setNumeProdus(String numeProdus) {
        this.numeProdus = numeProdus;
    }

    public void setCasier(String casier) {
        this.casier = casier;
    }

    public void setMetodaDePlata1(String metodaDePlata1) {
        this.metodaDePlata1 = metodaDePlata1;
    }

    public void setMetodaDePlata2(String metodaDePlata2) {
        this.metodaDePlata2 = metodaDePlata2;
    }

    public void setMetodaDePlata3(String metodaDePlata3) {
        this.metodaDePlata3 = metodaDePlata3;
    }

    public void setTimp(String timp) {
        this.timp = timp;
    }

    public int getNrBon() {
        return nrBon;
    }

    public int getNrBuc() {
        return nrBuc;
    }

    public double getRest() {
        return rest;
    }

    public double getPretUnitar() {
        return pretUnitar;
    }

    public double getPretTotal() {
        return pretTotal;
    }

    public String getNumeProdus() {
        return numeProdus;
    }

    public String getCasier() {
        return casier;
    }

    public String getMetodaDePlata1() {
        return metodaDePlata1;
    }

    public String getMetodaDePlata2() {
        return metodaDePlata2;
    }

    public String getMetodaDePlata3() {
        return metodaDePlata3;
    }

    public String getTimp() {
        return timp;
    }
    
   
    
    
            

  
  }

src/main/java/com/mycompany/casademarcat/ticheteDeMasa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.casademarcat;

import static com.mycompany.casademarcat.loginAproved.contorTabelColoane;
import static com.mycompany.casademarcat.loginAproved.contorTabelLinii;
import javax.swing.JSpinner;
import javax.swing.SpinnerNumberModel;
import javax.swing.table.DefaultTableModel;

/**
 *
 * @author kodie
 */
public class ticheteDeMasa extends javax.swing.JFrame  {
    int aux;
    /**
     * Creates new form ticheteDeMasa
     */
    public ticheteDeMasa() {
        initComponents();
        
        
//        jTextField2.setVisible(false);
//        jLabel2.setVisible(false);
        
    }

    /**
     * This method is called from within the constructor to initialize the form.
     * WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the Form Editor.
     */
    @SuppressWarnings("unchecked")
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {

        jLabel1 = new javax.swing.JLabel();
        jLabel2 = new javax.swing.JLabel();
        jButton1 = new javax.swing.JButton();
        jButton2 = new javax.swing.JButton();
        jFormattedTextField1 = new javax.swing.JFormattedTextField();
        jTextField1 = new javax.swing.JTextField();

        setDefaultCloseOperation(javax.swing.WindowConstants.DISPOSE_ON_CLOSE);
        setAlwaysOnTop(true);

        jLabel1.setFont(new java.awt.Font("Tahoma", 1, 24)); // NOI18N
        jLabel1.setText("Valoare Tichet:");

        jLabel2.setFont(new java.awt.Font("Tahoma", 1, 24)); // NOI18N
        jLabel2.setText("Nr. Tichete");

        jButton1.setText("OK");
        jButton1.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                jButton1ActionPerformed(evt);
            }
        });

        jButton2.setText("Inapoi");
        jButton2.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                jButton2ActionPerformed(evt);
            }
        });

        jFormattedTextField1.setHorizontalAlignment(javax.swing.JTextField.RIGHT);
        jFormattedTextField1.setFont(new java.awt.Font("Tahoma", 1, 24)); // NOI18N
        jFormattedTextField1.setSelectionEnd(aux);
        jFormattedTextField1.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                jFormattedTextField1ActionPerformed(evt);
            }
        });
        jFormattedTextField1.addKeyListener(new java.awt.event.KeyAdapter() {
            public void keyTyped(java.awt.event.KeyEvent evt) {
                jFormattedTextField1KeyTyped(evt);
            }
        });

        jTextField1.setFont(new java.awt.Font("Tahoma", 1, 24)); // NOI18N
        jTextField1.setHorizontalAlignment(javax.swing.JTextField.RIGHT);
        jTextField1.addCaretListener(new javax.swing.event.CaretListener() {
            public void caretUpdate(javax.swing.event.CaretEvent evt) {
                jTextField1CaretUpdate(evt);
            }
        });

        javax.swing.GroupLayout layout = new javax.swing.GroupLayout(getContentPane());
        getContentPane().setLayout(layout);
        layout.setHorizontalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(layout.createSequentialGroup()
                .addGap(110, 110, 110)
                .addComponent(jButton1, javax.swing.GroupLayout.PREFERRED_SIZE, 76, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addGap(41, 41, 41)
                .addComponent(jButton2, javax.swing.GroupLayout.PREFERRED_SIZE, 87, javax.swing.GroupLayout.PREFERRED_SIZE)
                .addContainerGap(javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE))
            .addGroup(layout.createSequentialGroup()
                .addGap(37, 37, 37)
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addComponent(jLabel1)
                    .addComponent(jLabel2, javax.swing.GroupLayout.PREFERRED_SIZE, 180, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addGroup(layout.createSequentialGroup()
                        .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                        .addComponent(jFormattedTextField1, javax.swing.GroupLayout.PREFERRED_SIZE, 205, javax.swing.GroupLayout.PREFERRED_SIZE))
                    .addGroup(layout.createSequentialGroup()
                        .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE)
                        .addComponent(jTextField1, javax.swing.GroupLayout.PREFERRED_SIZE, 194, javax.swing.GroupLayout.PREFERRED_SIZE)))
                .addContainerGap())
        );

        layout.linkSize(javax.swing.SwingConstants.HORIZONTAL, new java.awt.Component[] {jButton1, jButton2});

        layout.linkSize(javax.swing.SwingConstants.HORIZONTAL, new java.awt.Component[] {jFormattedTextField1, jTextField1});

        layout.setVerticalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(layout.createSequentialGroup()
                .addGap(36, 36, 36)
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(jLabel1, javax.swing.GroupLayout.PREFERRED_SIZE, 56, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(jTextField1, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addGap(63, 63, 63)
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.TRAILING)
                    .addComponent(jFormattedTextField1, javax.swing.GroupLayout.PREFERRED_SIZE, 58, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(jLabel2, javax.swing.GroupLayout.PREFERRED_SIZE, 58, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED, 24, Short.MAX_VALUE)
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(jButton1, javax.swing.GroupLayout.PREFERRED_SIZE, 32, javax.swing.GroupLayout.PREFERRED_SIZE)
                    .addComponent(jButton2, javax.swing.GroupLayout.PREFERRED_SIZE, 32, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addGap(29, 29, 29))
        );

        layout.linkSize(javax.swing.SwingConstants.VERTICAL, new java.awt.Component[] {jFormattedTextField1, jTextField1});

        pack();
    }// </editor-fold>//GEN-END:initComponents

    private void jButton2ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jButton2ActionPerformed
        dispose();
    }//GEN-LAST:event_jButton2ActionPerformed

    private void jButton1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jButton1ActionPerformed
  
    loginAproved.actualizeazaTotalTichete((Double.parseDouble(jTextField1.getText()) * Double.parseDouble(jFormattedTextField1.getText())) * -1, Double.parseDouble(jTextField1.getText()), Integer.parseInt(jFormattedTextField1.getText()));
    loginAproved.calculeazaTotal();
    dispose();
       
    }//GEN-LAST:event_jButton1ActionPerformed

    private void jFormattedTextField1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jFormattedTextField1ActionPerformed
        
    }//GEN-LAST:event_jFormattedTextField1ActionPerformed

    private void jFormattedTextField1KeyTyped(java.awt.event.KeyEvent evt) {//GEN-FIRST:event_jFormattedTextField1KeyTyped
     char c =  evt.getKeyChar();
           if(Character.isLetter(c) && !evt.isAltDown())
            evt.consume();
    }//GEN-LAST:event_jFormattedTextField1KeyTyped

    private void jTextField1CaretUpdate(javax.swing.event.CaretEvent evt) {//GEN-FIRST:event_jTextField1CaretUpdate
        jFormattedTextField1.setText("" + (int)(loginAproved.sum / Double.parseDouble(jTextField1.getText())));
        aux = Integer.parseInt(jFormattedTextField1.getText());
    }//GEN-LAST:event_jTextField1CaretUpdate

    /**
     * @param args the command line arguments
     */
    public static void main(String args[]) {
        /* Set the Nimbus look and feel */
        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
        /* If Nimbus (introduced in Java SE 6) is not available, stay with the default look and feel.
         * For details see http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html 
         */
        try {
            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
                if ("Nimbus".equals(info.getName())) {
                    javax.swing.UIManager.setLookAndFeel(info.getClassName());
                    break;
                }
            }
        } catch (ClassNotFoundException ex) {
            java.util.logging.Logger.getLogger(ticheteDeMasa.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (InstantiationException ex) {
            java.util.logging.Logger.getLogger(ticheteDeMasa.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (IllegalAccessException ex) {
            java.util.logging.Logger.getLogger(ticheteDeMasa.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
            java.util.logging.Logger.getLogger(ticheteDeMasa.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        }
        //</editor-fold>

        /* Create and display the form */
        java.awt.EventQueue.invokeLater(new Runnable() {
            public void run() {
                new ticheteDeMasa().setVisible(true);
              
            }
        });
    }

    // Variables declaration - do not modify//GEN-BEGIN:variables
    public static javax.swing.JButton jButton1;
    private javax.swing.JButton jButton2;
    public static javax.swing.JFormattedTextField jFormattedTextField1;
    private javax.swing.JLabel jLabel1;
    private javax.swing.JLabel jLabel2;
    private javax.swing.JTextField jTextField1;
    // End of variables declaration//GEN-END:variables
}


src/main/java/com/mycompany/casademarcat/schimbareParola.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.casademarcat;

import static com.mycompany.casademarcat.login.loggedUser;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.logging.Level;
import java.util.logging.Logger;
import javax.swing.JOptionPane;

/**
 *
 * @author kodie
 */
public class schimbareParola extends javax.swing.JFrame {

    /**
     * Creates new form schimbareParola
     */
    public schimbareParola() {
        initComponents();
    }

    /**
     * This method is called from within the constructor to initialize the form.
     * WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the Form Editor.
     */
    @SuppressWarnings("unchecked")
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {

        jPanel1 = new javax.swing.JPanel();
        jLabel1 = new javax.swing.JLabel();
        jLabel2 = new javax.swing.JLabel();
        jLabel3 = new javax.swing.JLabel();
        jLabel4 = new javax.swing.JLabel();
        jPasswordField1 = new javax.swing.JPasswordField();
        jPasswordField2 = new javax.swing.JPasswordField();
        jPasswordField3 = new javax.swing.JPasswordField();
        jButton1 = new javax.swing.JButton();
        jButton2 = new javax.swing.JButton();

        setDefaultCloseOperation(javax.swing.WindowConstants.EXIT_ON_CLOSE);
        setTitle("Schimbare Parola");

        jLabel1.setFont(new java.awt.Font("Times New Roman", 1, 18)); // NOI18N
        jLabel1.setText("Schimbare Parola");

        jLabel2.setFont(new java.awt.Font("Times New Roman", 1, 14)); // NOI18N
        jLabel2.setText("Parola veche");

        jLabel3.setFont(new java.awt.Font("Times New Roman", 1, 14)); // NOI18N
        jLabel3.setText("Parola noua");

        jLabel4.setFont(new java.awt.Font("Times New Roman", 1, 14)); // NOI18N
        jLabel4.setText("Reintrodu parola noua");

        jPasswordField1.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                jPasswordField1ActionPerformed(evt);
            }
        });

        jButton1.setText("Ok");
        jButton1.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                jButton1ActionPerformed(evt);
            }
        });

        jButton2.setText("Anulare");
        jButton2.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                jButton2ActionPerformed(evt);
            }
        });

        javax.swing.GroupLayout jPanel1Layout = new javax.swing.GroupLayout(jPanel1);
        jPanel1.setLayout(jPanel1Layout);
        jPanel1Layout.setHorizontalGroup(
            jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(javax.swing.GroupLayout.Alignment.TRAILING, jPanel1Layout.createSequentialGroup()
                .addContainerGap(100, Short.MAX_VALUE)
                .addComponent(jLabel1)
                .addGap(100, 100, 100))
            .addGroup(jPanel1Layout.createSequentialGroup()
                .addContainerGap()
                .addGroup(jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.TRAILING)
                    .addGroup(jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                        .addComponent(jLabel2)
                        .addComponent(jLabel3)
                        .addComponent(jLabel4))
                    .addComponent(jButton1))
                .addGroup(jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addGroup(jPanel1Layout.createSequentialGroup()
                        .addGap(24, 24, 24)
                        .addGroup(jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                            .addComponent(jPasswordField1, javax.swing.GroupLayout.PREFERRED_SIZE, 152, javax.swing.GroupLayout.PREFERRED_SIZE)
                            .addComponent(jPasswordField3, javax.swing.GroupLayout.PREFERRED_SIZE, 152, javax.swing.GroupLayout.PREFERRED_SIZE)
                            .addComponent(jPasswordField2, javax.swing.GroupLayout.PREFERRED_SIZE, 152, javax.swing.GroupLayout.PREFERRED_SIZE)))
                    .addGroup(jPanel1Layout.createSequentialGroup()
                        .addGap(35, 35, 35)
                        .addComponent(jButton2)))
                .addContainerGap(javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE))
        );

        jPanel1Layout.linkSize(javax.swing.SwingConstants.HORIZONTAL, new java.awt.Component[] {jButton1, jButton2});

        jPanel1Layout.setVerticalGroup(
            jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(jPanel1Layout.createSequentialGroup()
                .addContainerGap()
                .addComponent(jLabel1)
                .addGap(18, 18, 18)
                .addGroup(jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(jLabel2)
                    .addComponent(jPasswordField1, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addGap(18, 18, 18)
                .addGroup(jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addComponent(jLabel3)
                    .addComponent(jPasswordField2, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addGap(18, 18, 18)
                .addGroup(jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addComponent(jLabel4)
                    .addComponent(jPasswordField3, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE))
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED, 47, Short.MAX_VALUE)
                .addGroup(jPanel1Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
                    .addComponent(jButton1)
                    .addComponent(jButton2))
                .addGap(32, 32, 32))
        );

        javax.swing.GroupLayout layout = new javax.swing.GroupLayout(getContentPane());
        getContentPane().setLayout(layout);
        layout.setHorizontalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addComponent(jPanel1, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
        );
        layout.setVerticalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addComponent(jPanel1, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
        );

        pack();
    }// </editor-fold>//GEN-END:initComponents

    private void jButton1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jButton1ActionPerformed
       try {             
            Connection conn = DriverManager.getConnection("jdbc:derby://localhost:1527/magazinDB; create = true ", "robert", "test");
            Statement st =  conn.createStatement();
            if(String.valueOf(jPasswordField1.getPassword()).equals(login.loggedPass))
            {
                if(String.valueOf(jPasswordField2.getPassword()).equals(String.valueOf(jPasswordField3.getPassword()))){
                        st.execute("UPDATE CONTURI SET PAROLA = '"+ String.valueOf(jPasswordField3.getPassword())+ "' WHERE NUMECONT like '" + login.loggedUser +"' and PAROLA like '" + login.loggedPass + "'" );  
                        JOptionPane.showMessageDialog(null, "Parola lui " + loggedUser + " a fost actualizata.");  
                        login.loggedPass = String.valueOf(jPasswordField3.getPassword());
                        dispose();
                      
                }else{
                    JOptionPane.showMessageDialog(null, "Verificati parolele introduse!");
                     jPasswordField2.setText(""); 
                     jPasswordField3.setText(""); 
                }     
            }else{
                   JOptionPane.showMessageDialog(null, "Parola veche nu este corecta!");  
                   jPasswordField1.setText(""); 
                   }
        }   
         catch (SQLException ex) {
            Logger.getLogger(login.class.getName()).log(Level.SEVERE, null, ex);
        }
    }//GEN-LAST:event_jButton1ActionPerformed

    private void jButton2ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jButton2ActionPerformed
       dispose();
    }//GEN-LAST:event_jButton2ActionPerformed

    private void jPasswordField1ActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_jPasswordField1ActionPerformed
        // TODO add your handling code here:
    }//GEN-LAST:event_jPasswordField1ActionPerformed

    /**
     * @param args the command line arguments
     */
    public static void main(String args[]) {
        /* Set the Nimbus look and feel */
        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
        /* If Nimbus (introduced in Java SE 6) is not available, stay with the default look and feel.
         * For details see http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html 
         */
        try {
            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
                if ("Nimbus".equals(info.getName())) {
                    javax.swing.UIManager.setLookAndFeel(info.getClassName());
                    break;
                }
            }
        } catch (ClassNotFoundException ex) {
            java.util.logging.Logger.getLogger(schimbareParola.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (InstantiationException ex) {
            java.util.logging.Logger.getLogger(schimbareParola.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (IllegalAccessException ex) {
            java.util.logging.Logger.getLogger(schimbareParola.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
            java.util.logging.Logger.getLogger(schimbareParola.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        }
        //</editor-fold>

        /* Create and display the form */
        java.awt.EventQueue.invokeLater(new Runnable() {
            public void run() {
                new schimbareParola().setVisible(true);
            }
        });
    }

    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JButton jButton1;
    private javax.swing.JButton jButton2;
    private javax.swing.JLabel jLabel1;
    private javax.swing.JLabel jLabel2;
    private javax.swing.JLabel jLabel3;
    private javax.swing.JLabel jLabel4;
    private javax.swing.JPanel jPanel1;
    private javax.swing.JPasswordField jPasswordField1;
    private javax.swing.JPasswordField jPasswordField2;
    private javax.swing.JPasswordField jPasswordField3;
    // End of variables declaration//GEN-END:variables
}
